src/lib/common/JsonHelper.cpp

```diff
@@ -33,233 +34,242 @@
 #include <iomanip>
 
 
-/* ****************************************************************************
-*
-* toJsonString -
-*/
-std::string toJsonString(const std::string& input)
-{
-  std::ostringstream ss;
-
-  ss << '"';
-  for (std::string::const_iterator iter = input.begin(); iter != input.end(); ++iter)
+JsonHelper::JsonHelper(int _indent) : sb(), writer(sb), pwriter(sb) {
+  indent = _indent;
+  if (indent < 0)
   {
-    /* FIXME P3: This function ensures that if the DB holds special characters (which are
-     * not supported in JSON according to its specification), they are converted to their escaped
-     * representations. The process wouldn't be necessary if the DB couldn't hold such special characters, 
-     * but as long as we support NGSIv1, it is better to have the check (e.g. a newline could be 
-     * used in an attribute value using XML). Even removing NGSIv1, we have to ensure that the 
-     * input parser (rapidjson) doesn't inject not supported JSON characters in the DB (this needs to be
-     * investigated in the rapidjson documentation)
-     *
-     * JSON specification is a bit obscure about the need of escaping / (what they call 'solidus'). The
-     * picture at JSON specification (http://www.json.org/) seems suggesting so, but after a careful reading of
-     * https://tools.ietf.org/html/rfc4627#section-2.5, we can conclude it is not mandatory. Online checkers
-     * such as http://jsonlint.com confirm this. Looking in some online discussions
-     * (http://andowebsit.es/blog/noteslog.com/post/the-solidus-issue/ and
-     * https://groups.google.com/forum/#!topic/opensocial-and-gadgets-spec/FkLsC-2blbo) it seems that
-     * escaping / may have sense in some situations related with JavaScript code, which is not the case of Orion.
-     */
-    switch (char ch = *iter)
-    {
-    case '\\': ss << "\\\\"; break;
-    case '"': ss << "\\\""; break;    
-    case '\b': ss << "\\b"; break;
-    case '\f': ss << "\\f"; break;
-    case '\n': ss << "\\n"; break;
-    case '\r': ss << "\\r"; break;
-    case '\t': ss << "\\t"; break;
-    default:
-      /* Converting the rest of special chars 0-31 to \u00xx. Note that 0x80 - 0xFF are untouched as they
-       * correspond to UTF-8 multi-byte characters */
-      if (ch >= 0 && ch <= 0x1F)
-      {
-        static const char intToHex[16] =  { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' } ;
-
-        ss << "\\u00" << intToHex[(ch & 0xF0) >> 4] << intToHex[ch & 0x0F];
-      }
-      else
-      {
-        ss << ch;
-      }
-      break;
-    } //end-switch
-
-  } //end-for
-  ss << '"';
-  return ss.str();
+    indent = DEFAULT_JSON_INDENT;    
+  }
+  pwriter.SetIndent(' ', indent);
 }
 
 
+void JsonHelper::Null() {
```

Comment:
First of all, great job!

A minor detail, this line here (and many lines that follow), doesn't follow the Orion style guide (see https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/contribution_guidelines.md).

As you will see in that doc, and throughout the code, we put brackets (`{}`) on a line of its own, all functions have a function header and functions/methods are camelCase, starting with lowercase (`null`, `int`, etc may give problems with the compiler for being reserved words ... - if so, an exception-decision needs to be made, and added to the style guide).

Please read the style guide and fix these minor details.
[ I wont add a github-comment for each occurrence here, but instead do a re-check after this has been fixed ]

Reply:
ok, I will fix those problems.